paimon-core/src/main/java/org/apache/paimon/table/system/BucketsTable.java

```diff
@@ -181,9 +184,22 @@ public RecordReader<InternalRow> createReader(Split split) {
                     new RowDataToObjectArrayConverter(
                             fileStoreTable.schema().logicalPartitionType());
 
+            // sorted by partition and bucket
+            Map<Pair<String, Integer>, BucketEntry> pairBucketsMap =
+                    new TreeMap<>(
+                            Comparator.comparing((Pair<String, Integer> o) -> o.getLeft())
+                                    .thenComparing(Pair::getRight));
+            buckets.forEach(
```

Comment:
```
Function<BucketEntry, String> toPartString = e -> Arrays.toString(converter.convert(e.partition()));
buckets.sort(Comparator.comparing(toPartString).thenComparing(BucketEntry::bucket));
```

Reply:
A little difference, change `List<BucketEntry> buckets` to `List<Pair<String, BucketEntry>> bucketList`, to avoid `converter.convert(e.partition())` compute twice.